src/preload/preload.js

```diff
@@ -1,141 +1,387 @@
+   * @param {FormData=} formData
+   */
+  function sendXmlGetRequest(url, sourceNode, formData = undefined) {
+    const xhr = new XMLHttpRequest()
+    if (formData) {
+      url += '?' + new URLSearchParams(formData.entries()).toString()
     }
+    xhr.open('GET', url);
+    xhr.setRequestHeader("HX-Preloaded", "true")
+    xhr.onload = function() { processResponse(sourceNode, xhr.responseText) }
+    xhr.send()
+  }
+
+  /**
+   * Process request response by marking node `DONE` to prevent repeated
+   * requests, except if preload attribute contains `always`,
+   * and load linked resources (e.g. images) returned in the response 
+   * if `preload-images` attribute is `true`
+   * @param {Node} node 
+   * @param {string} responseText 
+   */
+  function processResponse(node, responseText) {
+    node.preloadState = node.preloadAlways ? 'READY' : 'DONE'
 
-    // Search for all child nodes that have a "preload" attribute
-    const parent = event.target || event.detail.elt;
-    parent.querySelectorAll("[preload]").forEach(function(node) {
-      // Initialize the node with the "preload" attribute
-      init(node)
+    if (getClosestAttribute(node, 'preload-images') === 'true') {
+      // Load linked resources
+      document.createElement('div').innerHTML = responseText
+    }
+  }
+
+  /**
+   * Gets attribute value from node or one of its parents
+   * @param {Node} node 
+   * @param {string} attribute 
+   * @returns { string | undefined }
+   */
+  function getClosestAttribute(node, attribute) {
+    if (node == undefined) { return undefined }
+    return node.getAttribute(attribute)
+      || node.getAttribute('data-' + attribute)
+      || getClosestAttribute(node.parentElement, attribute)
+  }
 
-      // Initialize all child elements that are anchors or have `hx-get` (use with care)
-      node.querySelectorAll('a,[hx-get],[data-hx-get]').forEach(init)
-    })
+  /**
+   * Determines if node is valid for preloading and should be
+   * initialized by setting up event listeners and handlers
+   * @param {Node} node 
+   * @returns {boolean}
+   */
+  function isValidNodeForPreloading(node) {
+    // Add listeners only to nodes which include "GET" transactions
+    // or preloadable "GET" form elements
+    const getReqAttrs = ['href', 'hx-get', 'data-hx-get'];
+    const includesGetRequest = node => getReqAttrs.some(a => node.hasAttribute(a))
+                                        || node.method === 'get';
+    const isPreloadableGetFormElement = node.form instanceof HTMLFormElement
+                                        && includesGetRequest(node.form)
+                                        && isPreloadableFormElement(node)
+    if (!includesGetRequest(node) && !isPreloadableGetFormElement) {
+      return false
+    }
+
+    // Don't preload <input> elements contained in <label>
+    // to prevent sending two requests
+    if (node.tagName instanceof HTMLInputElement && node.closest('label')) {
```

Comment:
Seems you forgot to remove the `tagName` access here, as the `instanceof` check is currently performed on a string rather than the node itself
(Also, the fact that this mistake doesn't fail any test in the current test suite, _might_ - unsure as I haven't reviewed the tests yet - reveal that a test case is missing to check this specific case?)

Reply:
@Telroshan Yes, I was too fast here. 😆  Fixed and added test for it in https://github.com/bigskysoftware/htmx-extensions/pull/106/commits/72c2fe25741ff8a16f084ca328bd97ac4cb58826.